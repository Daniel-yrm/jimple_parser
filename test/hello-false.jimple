public final class MainKt extends java.lang.Object
{

    public static final void main()
    {
        java.lang.AssertionError $r0;
        boolean $z0;

        $z0 = <kotlin._Assertions: boolean ENABLED>;

        if $z0 == 0 goto label1;

        $r0 = new java.lang.AssertionError;

        specialinvoke $r0.<java.lang.AssertionError: void <init>(java.lang.Object)>("Assertion failed");

        throw $r0;

     label1:
        return;
    }

    public static void main(java.lang.String[])
    {
        java.lang.String[] r0;

        r0 := @parameter0: java.lang.String[];

        staticinvoke <MainKt: void main()>();

        return;
    }
}
