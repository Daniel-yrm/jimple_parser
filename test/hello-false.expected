{"extends":"java.lang.Object","object":"Class","name":"MainKt","modifiers":["public","final"],"content":[{"object":"Method","name":"main","modifiers":["public","static","final"],"content":[{"object":"Variable","name":"$r0","type":{"identifier":"java.lang.AssertionError","dimensions":0}},{"object":"Variable","name":"$z0","type":{"identifier":"boolean","dimensions":0}},{"object":"SetVariable","name":"$z0","value":{"expr_type":"field_access","from":"kotlin._Assertions","type":{"identifier":"boolean","dimensions":0},"field":"ENABLED"}},{"goto":"label1","object":"If","expression":{"rhs":{"expr_type":"constant","value":"0"},"operator":"==","expr_type":"binop","lhs":{"expr_type":"symbol","value":"$z0"}}},{"object":"SetVariable","name":"$r0","value":{"expr_type":"new","type":{"identifier":"java.lang.AssertionError","dimensions":0}}},{"base_class":"java.lang.AssertionError","parameter_type":[{"identifier":"java.lang.Object","dimensions":0}],"object":"SpecialInvoke","method":"<init>","variable":"$r0","parameters":[]},{"expr":{"expr_type":"symbol","value":"$r0"},"object":"Throw"},{"label_id":"label1","object":"Label","content":[]},{"object":"Return"}],"type":{"identifier":"void","dimensions":0},"parameters":[]},{"object":"Method","name":"main","modifiers":["public","static"],"content":[{"object":"Variable","name":"r0","type":{"identifier":"java.lang.String","dimensions":1}},{"object":"SetVariable","name":"r0","value":{"to":{"identifier":"java.lang.String","dimensions":1},"expr_type":"cast","from":{"expr_type":"symbol","value":"@parameter0"}}},{"base_class":"MainKt","object":"StaticInvoke","method":"main","parameters":[]},{"object":"Return"}],"type":{"identifier":"void","dimensions":0},"parameters":[{"identifier":"java.lang.String","dimensions":1}]}]}